### PR TITLE
Tpm2 header cleanup

### PR DESCRIPTION
**Pull request by ddstreet**
Remove calls to dlopen_tpm2() from outside tpm2-util.c (except test), and remove libtss2 symbols from tpm-util.h; other code should use simplified functions provided by tpm2-util.c.

On top of #27627

### CODE REVIEW

**Comment by poettering** [comment]
also lgtm, once the other two PR are merged + and this one rebased.

**Comment by ddstreet** [comment]
rebased on #27627

**Review by DaanDeMeyer — APPROVED** [review]

**Pull request merged**
